core/src/test/java/io/aiven/klaw/TopicRequestValidatorImplIT.java

```diff
@@ -0,0 +1,276 @@
+import io.aiven.klaw.service.MailUtils;
+import io.aiven.klaw.service.TopicControllerService;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = UiapiApplication.class)
+@TestPropertySource(locations = "classpath:test-application-rdbms.properties")
+@AutoConfigureMockMvc
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DirtiesContext
+public class TopicRequestValidatorImplIT {
+
+  public static final String KWUSER = "kwuser";
+  @Autowired private Validator validator;
+
+  @MockBean private CommonUtilsService commonUtilsService;
+
+  @MockBean private TopicControllerService topicControllerService;
+
+  @MockBean private MailUtils mailService;
+
+  private UtilMethods utilMethods;
+
+  @Mock private UserDetails userDetails;
+
+  @BeforeEach
+  public void mockDefaults() {
+    utilMethods = new UtilMethods();
+    loginMock();
+  }
+
+  @Test
+  @Order(1)
+  public void isValidTestNotAuthorizedUser() {
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(true);
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString()).contains(ApiResultStatus.NOT_AUTHORIZED.value);
+  }
+
+  @Test
+  @Order(2)
+  public void isValidTestTenantFiltering() {
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn(KWUSER);
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("2"));
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
```

Comment:
```suggestion
    assertThat(violations).hasSize(1);
```
assertJ has `hasSize` it's better to use it here

Reply:
Agree, updated.